Code/MinimalLib/minilib.cpp

```diff
@@ -707,7 +640,8 @@ int JSSubstructLibrary::add_smiles(const std::string &smi) {
 }
 
 JSMol *JSSubstructLibrary::get_mol(unsigned int i) {
-  return new JSMol(new RWMol(*d_sslib->getMol(i)));
+  auto mol = new RWMol(*d_sslib->getMol(i));
+  return mol ? new JSMol(mol) : nullptr;
```

Comment:
I'm surprised missed this the first time through.
You're C background is showing through again. ;-)

This is a meaningless check since `new` always succeeds. The original code is exactly equivalent to this 

There's a bunch more of these below that should also be changed.

Reply:
Done. I also fixed a test that was still referencing `Reaction.is_valid()`, which was not exposed previously and still isn't.